extensions-contrib/rabbit-stream-indexing-service/src/main/java/org/apache/druid/indexing/rabbitstream/RabbitStreamIndexTaskIOConfig.java

```diff
@@ -50,8 +50,8 @@ public RabbitStreamIndexTaskIOConfig(
       @JsonProperty("consumerProperties") Map<String, Object> consumerProperties,
       @JsonProperty("pollTimeout") Long pollTimeout,
       @JsonProperty("useTransaction") Boolean useTransaction,
-      @JsonProperty("minimumMessageTime") DateTime minimumMessageTime,
-      @JsonProperty("maximumMessageTime") DateTime maximumMessageTime,
+      @JsonProperty("lateMessageRejectionPeriod") Duration lateMessageRejectionPeriod,
+      @JsonProperty("earlyMessageRejectionPeriod") Duration earlyMessageRejectionPeriod,
```

Comment:
These new fields need to be in addition to the old fields to ensure a newer version of the overlord works with older versions of the indexing tasks. But more importantly, in the case of a rollback - an older overlord needs to work with a newer version of indexing services

Reply:
May to avoid all these issues, How about instead of propagating periods, Is it okay if we just recalculate the period min/max time (and avoid introducing these new fields)?